### ISSUE TITLE
[FE] My VA: 508-defect-2 [COGNITION, SEMANTIC MARKUP]: Verify identity prompt should be designed and written semantically

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25050]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to navigate through logical headings that aren't skipped.
As a voice assistive tech user, I want links to be designed as links so I can activate them correctly.
As a keyboard user, I want links to be styled as links so that I know to use `enter` and not `space`.

## Details

- Headings issue was caught by axe and headingsMap
- Link is styled as a button

## Acceptance Criteria

- [ ] "Verify your identity" heading is a `h2`
- [ ] "Verify your identity" link (styled as a button currently) is styled as an action link

## Steps to Recreate

Log in as LOA 1, user A

## Screenshots or Trace Logs
<img width="1437" alt="Screen Shot 2021-05-20 at 5 02 24 PM" src="https://user-images.githubusercontent.com/14154792/119048719-98cc8680-b98d-11eb-9d5c-cf102315037e.png">

**Comment by joshkimux** [comment]
@allison0034 design wise, this alert wraps a little too early at smaller widths

**Comment by allison0034** [comment]
@joshkimux I could not agree more. These alerts are not mobile friendly and I hope they can be redesigned. @rtwell now that accessibility is flagging it, do you think we can do something about it? 

Does anyone know why there is 64px of right padding on the alerts? Seems a bit much compared to the 10px border and 24px padding on the left. Also, I 'quickly' put together 2 options and decreased the padding.
<img width="968" alt="Screen Shot 2021-05-21 at 8 35 12 AM" src="https://user-images.githubusercontent.com/72398073/119138091-8430d280-ba0f-11eb-9984-16f2a5c67c30.png">

**Comment by allison0034** [comment]
@joshkimux @rtwell there is also [another ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/14485) with mobile friendly options

**Comment by rtwell** [comment]
the challenge isnt so much that we dont have design solutions for something like this, but the engineering lift. these alerts are used across VA.gov, and it's my understanding they are not all pulling from the same code base. What team has the bandwidth to stop what they are doing and pick up this work? This is the design/tech debt problem that VA is starting to focus on this year. 

Additionally, I made [these updates](https://www.sketch.com/s/eb44dd9d-c8b7-40e7-802b-c8978eae53c8) for the system maintenance banners, which could easily be retrofitted to our current alerts back in May of 2020.

**Comment by andaleliz** [comment]
>which could easily be retrofitted to our current alerts.

Can you say more about that @rtwell? Specifically, how you would see those updates being applied easily to our current alerts? Introducing the same kind of padding and mobile adjustments you outlined in your spec sheet would definitely improve our other alert pattern on mobile.

**Comment by rtwell** [comment]
the version I designed is for full width— i don't think it would be a stretch to apply the spacing/typography to inline alerts

**Comment by andaleliz** [comment]
I don't think so either, and that would be awesome! As you know it's a struggle for VFS teams to focus on component design in the midst of project work. Since you were the designer of the full width alert - is that something you might be able to shepherd along for consideration of adoption into the design system? You outlined this process for me in a DM:

1. propose the thing
2. design council says yes/no
3. vfs team makes the thing (sketch or code or whatever), drop it on the experimental page
4. validate it
5. document and add it to the design system proper

There's enough research going on that I'm sure there's an alert being tested somewhere that a VFS can validate your design in before step 5 takes place. I know my team would have an opportunity to validate this in round 2 of our notifications research.

**Comment by andaleliz** [comment]
>FYSA, once Kevin and I publish the design system memo through Charles and Chris, it will be required for all teams working on new things to take these steps, and the work will be baked into sprint planning and project work.

My understanding has been that this has always been the way to go about working on new things, which is why I (and I think other designers) try to stick with what is in the design system.  But, that's good to know that it'll be more formalized. 

>this is mostly an engineering resource challenge

Oh, I must not be not totally clear on how that all works then. I would think that this could be implemented at the design system component level and improve the majority of the alerts on VA.gov.  I'm sure there are some custom implementations out there; was thinking those could be fixed as they are discovered by the team they belong to.

**Comment by rtwell** [comment]
nope, this has not been the way folks add/update new things. usually ppl make stuff and get it into production, with little-to-no consideration for the design system, documentation or otherwise. this is why production is so detached from the design system. formalizing and mandating this new process will hopefully change that over time. 

there is no question there are custom implementations, and also we have to consider the react components and the liquid templates. this is what the new web components should also help address — one codebase to rule them all rather than the 2 we have now.

**Comment by tressaellen** [comment]
@joshkimux - for this ticket we are planning on just fulfilling the acceptance criteria: 

- [ ] "Verify your identity" heading is a h2
- [ ] "Verify your identity" link (styled as a button currently) is styled as an action link

Does that work for this ticket?

As far as the design of the alert - I also agree that there is room for improvement on the padding etc, should we track that work elsewhere since it's out of scope for this ticket? @rtwell @allison0034

**Comment by joshkimux** [comment]
Yes, that does the trick 🥳

**Comment by tressaellen** [comment]
Here is a screenshot for reference for the action link. 
<img width="770" alt="Screen Shot 2021-08-09 at 10 39 40 AM" src="https://user-images.githubusercontent.com/58752772/128742127-a08084e0-7b80-48d8-9970-a463a8f718ac.png">

**Comment by ajakabcin** [comment]
Hey @joshkimux or @AngelaFowler82 could you take a look at this for validation? Thank you in advance!

**Comment by joshkimux** [comment]
<img width="698" alt="Screen Shot 2021-08-18 at 3 09 57 PM" src="https://user-images.githubusercontent.com/14154792/129957728-f2a4125d-4a41-428e-9154-48ed05ec982e.png">

Brilliant as always from @erikphansen ! Closing this out 🥳
